macOS/debugger.cpp

```diff
@@ -772,10 +794,32 @@ void Debugger::ExtractCodeRanges(void *base_address,
         continue;
       }
 
+#ifdef ARM64
+      if (strcmp(segment_cmd->segname, "__TEXT") == 0) {
+        mach_vm_address_t segment_start_addr = (mach_vm_address_t)segment_cmd->vmaddr + file_vm_slide;
+        mach_vm_address_t segment_end_addr = (mach_vm_address_t)segment_cmd->vmaddr + file_vm_slide + segment_cmd->vmsize;
+        AddressRange arm_ar;
+        arm_ar.from = segment_start_addr;
+        arm_ar.to = segment_end_addr;
+        size_t range_size = arm_ar.to - arm_ar.from;
+        arm_ar.data = (char *)malloc(range_size);
+        RemoteRead((void*)arm_ar.from, arm_ar.data, range_size);
+
+        ExtractSegmentCodeRanges(segment_start_addr, segment_end_addr, executable_ranges, code_size);
```

Comment:
Why is it still needed to call ExtractSegmentCodeRanges and delete all ranges it extracted immediately?

Reply:
Yeah, this is a bit hacky - the reason is that ExtractSegmentCodeRanges marks the pages as read-only and since this requires a bit more logic than just calling mach_vm_protect, I still call that function.

Reply:
Ah, I see. It would be useful to include a comment explaining that, as I'm sure I'll forget about it and ask the same question later :-) Would also be useful to have a comment for this whole `#ifdef ARM64` block.